microbench/src/main/java/io/netty/microbench/headers/HeadersBenchmark.java

```diff
@@ -139,4 +163,546 @@ public void http2Iterate(Blackhole bh) {
             bh.consume(entry);
         }
     }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void httpAddAllFastest(Blackhole bh) {
+        bh.consume(new DefaultHttpHeaders(false).add(httpHeaders));
```

Comment:
That's a good point. That would require to clear the headers before each invocation though. Something like

```java
@Setup(Level.Invocation)
public void setup() {
   addAllFastestHeaders.clear();
}

@Benchmark
public void httpAddAllFastest() {
 addAllFastestHeaders.add(httpHeaders);  
}

```

Would have to test how this works out. Allocations would be less but not much less, as we mostly allocate `HeaderEntry` objects and those would also be there in `clear()`.

Adding `-verbose:gc` to the JVM flags, opens one's eyes to much GC is going on during a benchmark.